openlibrary/plugins/openlibrary/code.py

```diff
@@ -1243,6 +1243,22 @@ def setup_template_globals():
         get_cover_url,
     )
 
+    SUPPORTED_LANGUAGES = {
+        "cs": {"localized": 'Czech', "native": "Čeština"},
```

Comment:
These will need to be wrapped with the `_` in order to be actually localized, e.g.:

```
{"localized": _('Czech'), "native": "Čeština"}
```

Otherwise they will always be in English.

This also means we can't have this be a constant global, or else the `_` will only read the user language once! So we'll want to make this into a function so that `_` can get the user language per-request. E.g.

```py
def get_supported_languages():
    return {
        "cs": {"localized": _('Czech'), "native": "Čeština"},
        ...
    }
```

And then make this function publicly accessible from the template.

Reply:
That makes sense! Initially when I tried to create the dictionary, it followed this format:
```
{"localized": _('Czech'), "native": "Čeština"}
```
 but I kept getting an error message and so I left the dictionary in the format it currently is without the underscore with the intent of using the underscore in whatever page it's imported into. In PR #9889 I actually use the localization `_`  [here](https://github.com/internetarchive/openlibrary/compare/master...ChinoUkaegbu:openlibrary:refactor/language-list#diff-50a00e3deded6b6cac0418c1a08e5449660cf1f9115375872602f16a98b480afR5) and it works so I'm wondering if I should follow that format in this case or just have that `get_supported_languages()` dictionary with the localization embedded in

Reply:
Ah you will have to import the `_` in the python file to get access to it, with `from openlibrary.i18n import gettext as _`! That should do the trick.

Unfortunately `_` only works with static text, since essentially what happens is _before_ the code is run, the python and html is analyzed and all `_("...")` are extracted and put into a separate file for our translators. This means that `_` doesn't work with variables, only literal strings.

Reply:
The import worked, thank you! I've addressed the concerns (hopefully) in #9906

Reply:
Wonderful thank you! I responded there 😊